### PR TITLE
Refs #23919 -- Removed Python 2 proxied methods on File.

### PR DESCRIPTION
**Pull request by jschneier**
I noticed this when looking at 98ee57e343206ef553de78b22be5e9a6dacb5060 again. Both `xreadlines` and `softspace` used to exist on file objects but were removed in Python 3. From taking a critical look at what counts as a file I'd say `.newlines` is close but I have left it for now, these 2 should be non-controversial.

### CODE REVIEW

**Comment by timgraham** [comment]
I don't think it's best to remove `HttpRequest.xreadlines()` without a deprecation considering that method works fine on Python 3.

**Comment by jschneier** [comment]
You're probably right, I updated my patch.

File: django/http/request.py [hunk 1]
```diff
@@ -328,14 +330,20 @@ def readline(self, *args, **kwargs):
...
             yield buf
 
-    __iter__ = xreadlines
+    def xreadlines(self):
+        warnings.warn(
+            'The xreadlines() method is deprecated in favor of iterating the '
+            'request directly.', RemovedInDjango30Warning
+        )
+        for line in self:
+            yield line
```
Comment (charettes) [inline django/http/request.py]: use `yield from self`.

Comment (jschneier) [inline django/http/request.py]: done

File: docs/spelling_wordlist [hunk 2]
```diff
@@ -855,6 +856,7 @@ wsgi
 www
 xe
 xgettext
+xreadlines
```
Comment (timgraham) [inline docs/spelling_wordlist]: Shouldn't be needed in the word list as long as all instances of this are surrounded by backticks.

File: docs/spelling_wordlist [hunk 3]
```diff
@@ -14,6 +14,7 @@ alphanumerics
 amet
 amongst
 analytics
+api
```
Comment (timgraham) [inline docs/spelling_wordlist]: shouldn't be needed in the word list-- use API in the docs instead

File: docs/ref/request-response.txt [hunk 4]
```diff
@@ -378,7 +385,6 @@ Methods
         for element in ET.iterparse(request):
             process(element)
 
-
```
Comment (timgraham) [inline docs/ref/request-response.txt]: please don't make unrelated whitespace changes

**Comment by jschneier** [comment]
Addressed all concerns

**Comment by jschneier** [comment]
Ah need to create a separate ticket for the deprecation.

**Comment by jschneier** [comment]
Okay I have updated this PR to not handle the deprecation and will open a new ticket + PR
